lib/Window.js

```diff
@@ -23,105 +23,204 @@ module.exports = (Item) => {
     }
 
     acceptClick (click) {
-      assert.ok(click.mouseButton === 0 || click.mouseButton === 1)
+      const { mode, slot, mouseButton } = click
+      assert.ok(
+        (mode <= 6 && mode >= 0) &&
+        (mouseButton >= 0 && mouseButton <= 8) &&
+        ((slot >= 0 && slot < this.inventoryEnd) || slot === -999),
+        'invalid operation')
+
+      switch (click.mode) {
+        case 0:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.mouseClick(click)
+          break
+
+        case 1:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.shiftClick(click)
+          break
+
+        case 2:
+          assert.ok(mouseButton <= 8, 'invalid operation')
```

Comment:
You're ignoring mode 2, button 40, which is offhand swap key

Reply:
in minecraft 1.8 there is no offhand swap key.
i just implemented clicks for 1.8, as i have no experience with minecraft above version 1.8 and don't feel confident doing implementing that

Reply:
I don't consider this as a blocker